sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -32,7 +32,7 @@ object ParserUtils {
   /** Get the command which created the token. */
   def command(ctx: ParserRuleContext): String = {
     val stream = ctx.getStart.getInputStream
-    stream.getText(Interval.of(0, stream.size()))
+    stream.getText(Interval.of(0, stream.size() - 1))
```

Comment:
Is this a bug fix?

Reply:
Yes - the RHS of the interval should correspond to the last char we want `getText()` to return. the previous stream implementation would silently bound the RHS of the interval to the last char in the stream, but the new implementation can fail if the RHS is off the end of the stream. The interface documentation says that this shouldn't be considered a bug in ANTLR (even though it's pretty unfriendly).